app/Services/PayPalService.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Services;
+use Illuminate\Http\Request;
+use Laravel\Cashier\Subscription;
+use Carbon\Carbon;
+use Srmklive\PayPal\Services\PayPal as PayPalClient;
+
+class PayPalService
+{   
+    /**
+    * @param Request $request
+    * @return mixed
+    */
+   public function process(Request $request)
+   {
+        $user = $request->user();
```

Comment:
We also want to confirm that the user isn't already subbed, just to avoid some very angry customers and confused database (in case they open the page in two tabs)